src/leveling.ts

```diff
@@ -0,0 +1,209 @@
+import { EmbedBuilder } from 'discord.js';
+import { getDB, getDBList } from '@/db';
+import { User } from '@/models/users';
+import { sendEventLogMessage } from '@/util';
+import type { GuildMember, Message } from 'discord.js';
+
+export async function handleLeveling(message: Message): Promise<void> {
+	const levelingChannelBlacklist = getDBList('leveling.channels-blacklist');
+	if (levelingChannelBlacklist.includes(message.channelId)) {
+		return;
+	}
+
+	const supporterRoleID = getDB().get('roles.supporter');
+	const supporterRole = supporterRoleID && (await message.guild?.roles.fetch(supporterRoleID));
+	if (!supporterRole) {
+		console.log('Missing supporter role!');
+		return;
+	}
+	const trustedRoleID = getDB().get('roles.trusted');
+	const trustedRole = trustedRoleID && (await message.guild?.roles.fetch(trustedRoleID));
+	if (!trustedRole) {
+		console.log('Missing trusted role!');
+		return;
+	}
+
+	const xpRequiredForTrusted = parseInt(getDB().get('leveling.xp-required-for-trusted') ?? '0');
+	if (!xpRequiredForTrusted) {
+		console.log('Missing amount of XP required for trusted role!');
+		return;
+	}
```

Comment:
If `xpRequiredForTrusted` resolves to `0` (which it will if `leveling.xp-required-for-trusted` is not set), then this will trigger the return. Is that intended? This happens elsewhere, so I'm asking for all occurances of this

Reply:
It was kind of intended because I didn't think 0 doesn't make sense as a setting for any of these. But that may be a bit too opinionated.